default/python/universe_generation/galaxy.py

```diff
@@ -1,35 +1,349 @@
+        be added to enforce MAX_STARLANE_LENGTH"
+    max_dist = universe_tables.MAX_STARLANE_LENGTH
+    min_dist = universe_tables.MIN_SYSTEM_SEPARATION
+    p1_p2_dist = util.distance(p1, p2)
+    if p1_p2_dist < max_dist:
+        return []
+
+    # Pick a random point in an 2:1 ratio ellipse and then rotate and slide it in between p1 and p2
+    p1_p2_theta = acos((p2[0] - p1[0]) / p1_p2_dist)
+    p1_p2_scale = (p1_p2_dist - 2 * min_dist)/2
+    p1_p2_translation = ((p1[0] + p2[0])/2, (p1[1] + p2[1])/2)
+
+    radius_0space = uniform(0.0, 1.0)
+    angle_0space = uniform(0.0, 2.0 * pi)
+    rotated_point = (radius_0space * p1_p2_scale * cos(angle_0space + p1_p2_theta),
+                     radius_0space * p1_p2_scale * sin(angle_0space + p1_p2_theta))
+
+    p3 = (rotated_point[0] + p1_p2_translation[0], rotated_point[1] + p1_p2_translation[1])
+
+    return [p3] + stitching_positions(p1, p3) + stitching_positions(p2, p3)
+
+
+def enforce_max_distance(positions, adjacency_grid):
+    """
+    Adds extra positions between groups of positions to guarrantee
+    that no groups of positions are separated from all other groups
+    of positions by more than universe_tables.MAX_STARLANE_LENGTH
+
+    Find all groups of positions, where every member is within
+    MAX_STARLANE_LENGTH of at least one other member of the group,
+    but all members of the group are more than MAX_STARLANE_LENGTH
+    from all positions in other groups.
+
+    For each group:
+      - Remove it from the adjacency grid and find its nearest neighboring position.
+      - Add positions between the nearest neighbor and the group to
+        ensure every position has at least one neighbor within
+        MAX_STARLANE_LENGTH.
+    """
+    # Find all clusters
+    clusterer = Clusterer(positions, adjacency_grid)
+
+    if len(clusterer) == 1:
+        print "All systems positioned in a single connected cluster"
+    else:
+        print "{} clusters separated by more the MAX_STARLANE_LENGTH.  Starting to fill gaps.".format(len(clusterer))
+
+    while len(clusterer) > 1:
+        smallest_cluster = clusterer.smallest_isolated_cluster()
+        print "Searching for nearest neighbor position to a cluster with {} positions.".format(len(smallest_cluster))
+        for pos in smallest_cluster:
+            adjacency_grid.remove_pos(pos)
+        # Find nearest neighbour
+        (_, p1, p2) = min([(util.distance(pos, nn), pos, nn)
+                           for nn in adjacency_grid.nearest_neighbor(pos)
+                           for pos in smallest_cluster])
+
+        for pos in smallest_cluster:
+            adjacency_grid.insert_pos(pos)
+        # Add extra positions
+        extra_positions = stitching_positions(p1, p2)
+        for i_extra, p3 in enumerate(extra_positions):
+            print "Adding {} of {} extra positions at {} to enforce max spacing between {} and {}."\
+                .format(i_extra, len(extra_positions), p3, p1, p2)
```

Comment:
`i_extra + 1`, otherwise you get log messages like this:
```
Adding 0 of 1 extra positions at (1329.2735707173076, 599.3023108153775) to enforce max spacing between (1311.4786665332913, 636.6901601665741) and (1353.0400945377253, 562.9319676906877).
```
I assume this should read "Adding 1 of 1 extra positions..."